### PR TITLE
Killed cons->truecolor in favor of cons->color

### LINKED ISSUE #9484: scr.color must be int instead of bool and replace scr.truecolor
This will simplify some more code related to colorhandling

* 0 : no color
* 1 : 16 colors
* 2 : 256 colors
* 3 = 16 milion colors

### PR DESCRIPTION
**Pull request by xarkes**
Fixes #9484

### CODE REVIEW

**Comment by XVilka** [comment]
It is 16, 256 and 16M no RGB.

File: libr/cons/rgb.c [hunk 1]
```diff
@@ -181,15 +181,15 @@ R_API char *r_cons_rgb_str(char *outstr, ut8 r, ut8 g, ut8 b, ut8 a) {
...
 		return outstr;
 	}
-	switch (r_cons_singleton ()->truecolor) {
-	case 1: // 256 color palette
+	switch (r_cons_singleton ()->color) {
+	case COLORS_RGB: // 256 color palette
 		sprintf (outstr, "\x1b[%d;5;%dm", fgbg, rgb (r, g, b));
 		break;
-	case 2: // 16M - xterm only
+	case COLORS_TRUECOLOR: // 16M - xterm only
```
Comment (XVilka) [inline libr/cons/rgb.c]: please remove reference "xterm only" - it is now almost everywhere

**Comment by xarkes** [comment]
Yes you're right, I changed it.

**Comment by xarkes** [comment]
Don't merge now, I'm gonna add something which might break the tests.

File: doc/hud [hunk 2]
```diff
@@ -6,7 +6,9 @@ analyze function	af
 analyze preludes	ap
 continue process execution	dc;?i
 disable colors	e scr.color=0
-enable colors	e scr.color=1
+enable colors (ANSI)	e scr.color=1
+enable colors (RGB)	e scr.color=2
```
Comment (XVilka) [inline doc/hud]: 256

File: doc/fortunes.tips [hunk 3]
```diff
@@ -38,7 +38,7 @@ Step through your seek history with the commands 'u' (undo) and 'U' (redo)
 Use hasher to calculate hashes of portion blocks of a file
 Use zoom.byte=entropy and press 'z' in visual mode to zoom out to see the entropy of the whole file
 Use 'zoom.byte=printable' in zoom mode ('z' in Visual mode) to find strings
-Set color to your screen with 'e scr.color=true'
+Set color to your screen with 'e scr.color=1' or truecolor with 'e scr.color=3'
```
Comment (XVilka) [inline doc/fortunes.tips]: "Add color to your screen with 'e scr.color=X' where 1 is 16 colors, 2 is 256 colors and 3 is 16M colors'

File: libr/core/cconfig.c [hunk 4]
```diff
@@ -2788,8 +2780,7 @@ R_API int r_core_config_init(RCore *core) {
 	SETCB ("scr.prompt", "true", &cb_scrprompt, "Show user prompt (used by r2 -q)");
 	SETCB ("scr.tee", "", &cb_teefile, "Pipe output to file of this name");
 	SETPREF ("scr.seek", "", "Seek to the specified address on startup");
-	SETICB ("scr.truecolor", 0, &cb_truecolor, "Manage color palette (0: ansi 16, 1: 256, 2: 16M)");
-	SETCB ("scr.color", (core->print->flags&R_PRINT_FLAGS_COLOR)?"true":"false", &cb_color, "Enable colors");
+	SETICB ("scr.color", (core->print->flags&R_PRINT_FLAGS_COLOR)?COLORS_16:COLORS_DISABLED, &cb_color, "Enable colors (0: none, 1: ansi, 2: rgb, 3: truecolor)");
```
Comment (XVilka) [inline libr/core/cconfig.c]: not rgb, 256 too.

File: libr/include/r_cons.h [hunk 5]
```diff
@@ -404,7 +406,7 @@ typedef struct r_cons_t {
 	 * current window. If NULL or "" no pager is used. */
 	char *pager;
 	int blankline;
-	int truecolor; // 0 = ansi, 1 = rgb (256), 2 = truecolor (16M)
+	int color; // 0 = none, 1 = ansi, 2 = rgb (256), 3 = truecolor (16M)
```
Comment (XVilka) [inline libr/include/r_cons.h]: maybe change rgb to "palette"?

File: binr/radare2/radare2.c [hunk 6]
```diff
@@ -521,11 +521,10 @@ int main(int argc, char **argv, char **envp) {
 			break;
 		case '0':
 			zerosep = true;
-			//r_config_set (r.config, "scr.color", "false");
 			/* implicit -q */
 			r_config_set (r.config, "scr.interactive", "false");
 			r_config_set (r.config, "scr.prompt", "false");
-			r_config_set (r.config, "scr.color", "false");
+			r_config_set_i (r.config, "scr.color", COLORS_DISABLED);
```
Comment (radare) [inline binr/radare2/radare2.c]: i would say COLOR_MODE_DISABLED, so this way follows a correct naming convention for bindings and such

Comment (xarkes) [reply binr/radare2/radare2.c]: then why do you merge haha

**Comment by radare** [comment]
Because there r more things to improve in rcons and that pr was good :)

**Pull request merged**